subxt/examples/legacy_reconnecting_rpc_client.rs

```diff
@@ -0,0 +1,71 @@
+//! Example to utilize the `reconnecting rpc client` in subxt
+//! which hidden behind behind `--feature unstable-reconnecting-rpc-client`
+//!
+//! To utilize full logs from the RPC client use:
+//! `RUST_LOG="jsonrpsee=trace,reconnecting_jsonrpsee_ws_client=trace"`
+
+#![allow(missing_docs)]
+
+use std::sync::Arc;
+use std::time::Duration;
+
+use subxt::backend::legacy::LegacyBackend;
+use subxt::backend::rpc::reconnecting_rpc_client::{Client, RetryPolicy};
+use subxt::backend::rpc::RpcClient;
+use subxt::config::Header;
+use subxt::error::{Error, RpcError};
+use subxt::{OnlineClient, PolkadotConfig};
+
+// Generate an interface that we can use from the node's metadata.
+#[subxt::subxt(runtime_metadata_path = "../artifacts/polkadot_metadata_small.scale")]
+pub mod polkadot {}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    tracing_subscriber::fmt::init();
+
+    // Create a new client with with a reconnecting RPC client.
+    let rpc = Arc::new(
+        Client::builder()
+            // Reconnect with exponential backoff
+            .retry_policy_for_reconnect(
+                RetryPolicy::exponential(Duration::from_millis(100))
+                    .with_max_delay(Duration::from_secs(10))
+                    .with_max_retries(usize::MAX),
+            )
+            .build("ws://localhost:9944".to_string())
+            .await?,
+    );
+
+    let backend: LegacyBackend<PolkadotConfig> =
+        LegacyBackend::builder().build(RpcClient::new(rpc.clone()));
+
+    let api: OnlineClient<PolkadotConfig> = OnlineClient::from_backend(Arc::new(backend)).await?;
+
+    // Print when the RPC client reconnects.
+    tokio::spawn(async move {
+        loop {
+            rpc.on_reconnect().await;
+            println!("RPC client reconnect initiated");
```

Comment:
Nit: when we get to this line I guess it's reconnected already? :)

Reply:
well this future returns when client starts to "reconnect" but I can change the APIs to more clean

Reply:
Aah ok; I assumed that `on_reconnect.await` would mean reconnection has finished by the time the await ends :)